### ISSUE TITLE
StorageFormat

### ISSUE THREAD

**Issue by muchino** [bsorrentino/maven-confluence-plugin#108]
Is there already a way or a plan to support the storage format instead of the old wiki syntax too? 
Reason one for my question: At the moment I have some difficulties to define the default template in our archetype. Because I like to use the pageproperties macro of confluence to generate automatic overview pages about our projects. 

But I have no lick with the old wiki style.

Second reason:  My teammates didn't known the old style, but we like to use and to add some macros per default, for example the jira macro. And if they use for example the Confluence Source Editor, they can create the documentation in their local Confluence (for the static stuff) and copy paste it to the maven project and extend this with the variables.

**Comment by bsorrentino** [comment]
Hi, thx for feedback

This was already requested [issue#47](https://github.com/bsorrentino/maven-confluence-plugin/issues/47)  but I didn't fixed because I considered wiki old style enough for mostly documentation cases

Now I'll evaluate adding such feature,  again

**Comment by bsorrentino** [comment]
Currently seems that it is not possible publish StorageFormat using xml-rpc api, so such feature should be implemented on milestone 5.0 [refer to issue 96](https://github.com/bsorrentino/maven-confluence-plugin/issues/96)

**Comment by muchino** [comment]
ok - but thank you for the test  :)

2016-06-23 9:18 GMT+02:00 bsorrentino <notifications@github.com>:

**Comment by bsorrentino** [comment]
I've submitted a topic to [Atlassian Answer](https://answers.atlassian.com/questions/39152954/is-it-possible-publish-storageformat-from-xml-rpc-api).
If there is a way i'll be able to add such **interesting** feature very quickly otherwise I've to move on REST API but this will take longer

**Comment by bsorrentino** [comment]
Hi I've released  version `5.0-SNAPSHOT`  with support for **storage format** it is enough that you use file extensions either **.xml** or **.xhtml**

Please, take a chance to test it and let me know

**Comment by bsorrentino** [comment]
Support added to release `5.0-beta1`. [See Usage Details](http://bsorrentino.github.io/maven-confluence-plugin/storageformat_guide.html)

**Comment by xanaxan** [comment]
![2017-04-21_15h51_48](https://cloud.githubusercontent.com/assets/2856284/25280995/27c25aa0-26ac-11e7-8baf-f9a26193fb85.png)

**Comment by bsorrentino** [comment]
Could you provide me the site descriptor ?

**Comment by xanaxan** [comment]
I've explained it in the gitter channel. The problem was 'using the confluence storage format (xhtml) not as a child element', but as the main page instead... Like in the <home> uri... where it obviously gets escaped. So we would like to define a parent confluence page (e.g. in the pom) and then list our jira issues in the 'current' page.

**Comment by bsorrentino** [comment]
Thanks for detailed explanation

Probably the problem is related to home page

I'll check that

**Comment by xanaxan** [comment]
Thank you very much for looking into it.
And with 'Like in the uri... ' i meant: in the uri attribute of the home tag in the site.xml

**Comment by bsorrentino** [comment]
I've deployed fix in `5.0-SNAPSHOT`

Please test it and let me know

Thanks in advance

**Note:**

> **how to  configure snapshot repository**
```xml
<pluginRepositories>
   <!-- IF YOU WANT STAY TUNED ON UPDATE REMOVE COMMENT -->
    <pluginRepository>
        <id>sonatype-repo</id>
        <url>https://oss.sonatype.org/content/repositories/snapshots</url>
        <releases>
            <enabled>false</enabled>
        </releases>
        <snapshots>
            <enabled>true</enabled>
        </snapshots>
    </pluginRepository>
</pluginRepositories>
```

**Comment by xanaxan** [comment]
Just tested it and everything works fine by using <home uri="***.xhtml">. Exactly like we needed it! 
Thank you very much!

**Comment by xanaxan** [comment]
but i still have to ask: is this going to be in an -rc5 in the near future or will it be included in the final version with the REST-Api-support?

**Comment by bsorrentino** [comment]
Thanks for feedback

I'm going to release another rc because I haven't complete all REST replacement yet

**Comment by bsorrentino** [comment]
fix released with version `5.0-rc5`
